### PR TITLE
Bug 1874771 - Only run 'github-push' Taskcluster tasks against main a…

### PR DESCRIPTION
**Pull request by ahal**
…nd release branches

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1874771

### CODE REVIEW

**Review by jcristau — CHANGES_REQUESTED** [review]
This will prevent merging PRs created by github actions.

**Comment by ahal** [comment]
> This will prevent merging PRs created by github actions.

How so? If a PR is created, then that PR will still get a `github-pull-request` graph. Currently those PRs are getting both `github-push` and `github-pull-request` graphs, and there's a race condition over which task ends up getting displayed in the Github checks UI (the later task overwrites the earlier task in the UI)

**Comment by ahal** [comment]
Ftr, I'm not opposed to leaving tasks on as level 1.. but in my experience with VPN, if your workflow involves creating pull-requests off of branches in the canonical repo, you probably want to disable `on-push` tasks in those branches

**Comment by jcristau** [comment]
> > This will prevent merging PRs created by github actions.
> 
> How so? If a PR is created, then that PR will still get a `github-pull-request` graph. Currently those PRs are getting both `github-push` and `github-pull-request` graphs, and there's a race condition over which task ends up getting displayed in the Github checks UI (the later task overwrites the earlier task in the UI)

I'll try to break down the different cases:
- for PRs created by collaborators, the pull-request graph is just fine.  We can get rid of any push graph here. (IMO they should push to a fork instead of the main repo, but that is not actually enforced)
- PRs created by humans that aren't collaborators are necessarily coming off of a fork.  They don't run ui-tests, they don't get a push graph.  No change there from this patch.
- PRs created by github actions is where we need to be careful.  These are on a branch on the main repo.  Their PR graph is incomplete, because it's not authored by a collaborator. We currently rely on the push graph in this case to satisfy merge conditions.  So we do need to keep that graph running here, that's why I suggested on the bug to add those branch name patterns both to the .tc.yml filter and to branch protections, although I'm not sure if we can say "github actions can push here but not human users".

**Comment by ahal** [comment]
Ah ok, thanks for the explanation. And I missed your comment on the bug. I just replied over there.

File: .taskcluster.yml [hunk 1]
```diff
@@ -147,7 +147,7 @@ tasks:
                   $if: >
                       tasks_for in ["action", "cron"]
                       || (isPullRequest && pullRequestAction in ["opened", "reopened", "synchronize"])
-                      || (tasks_for == "github-push" && head_ref[:10] != "refs/tags/" && short_head_ref != "staging.tmp" && short_head_ref != "trying.tmp" && short_head_ref[:8] != "mergify/")
+                      || (tasks_for == "github-push" && (short_head_ref == "main" || short_head_ref[:9] == "releases_"))
```
Comment (hneiva) [inline .taskcluster.yml:150]: I'm assuming "releases_*" branches are protected in this repo?

**Review by hneiva — APPROVED** [review]

**Review by hneiva — CHANGES_REQUESTED** [review]
I'll leave as requested changes according to previous conversation

**Pull request closed without merging**